main.py

```diff
@@ -16,6 +16,29 @@
     3: logging.DEBUG
 }
 
+class DictParamType(click.ParamType):
```

Comment:
Yes. This is annoying to have to write.

I think a lot of libraries punt on it because you really want it to be escapable, and that gets ugly quickly. I didn't make it escapable.